src/Expander.hs

```diff
@@ -1301,22 +1310,28 @@ expandOneForm prob stx
           implV <- Env.lookupVal transformerName <$> currentTransformerEnv
           case implV of
             Just (ValueClosure macroImpl) -> do
-              macroVal <- inEarlierPhase $ expandEval $
-                          apply macroImpl $
-                          ValueSyntax $ addScope p stepScope stx
+              macroVal <- inEarlierPhase
+                          $ pure
+                          $ apply macroImpl
+                          $ ValueSyntax
+                          $ addScope p stepScope stx
               case macroVal of
-                ValueMacroAction act -> do
-                  res <- interpretMacroAction prob act
-                  case res of
+                Left err -> throwError
+                            $ ValueNotMacro
```

Comment:
I don't think this should be a `ValueNotMacro` error. In this branch, looking up `transformerName` did successfully find a macro definition of type `(-> Syntax (Macro Syntax))`. That function was then applied to a `Syntax` argument, and the evaluation of that function failed for some reason, perhaps because `(error 'msg)` was called.

Reply:
Ha! The existing code already misuses `ValueNotMacro`. When expanding the macro call site `(my-macro)`, many things can go wrong:
1. `my-macro` might not be in scope, in which case `resolve` (line L1201) throws `Unknown` and the user sees `Unknown: my-macro`.
2. `my-macro` resolves to `EUserMacro transformerName`, but that `transformerName` is somehow not the `MacroVar` which was added to the transformer environment by `(define-macros ([my-macro ...]))` at the same time as `my-macro` was bound to `EUserMacro` in the binding table. For example, maybe `transformerName` is not in the binding table at all. This is a bug in Klister, so `InternalError` is thrown; `error` would have been fine too.
3. Same thing, but `transformerName` is bound to something other than a `(-> Syntax (Macro Syntax))`. This is also a bug in Klister, but `ValueNotMacro` is thrown. This is definitely wrong, since the error message associated with `ValueNotMacro` is "Not a macro monad value", but it's not even a `(Macro Syntax)` which is expected, it's a `(-> Syntax (Macro Syntax))`.
4. `transformerName` is bound to a value of type `(-> Syntax (Macro Syntax))`, but applying it to a `Syntax` value fails before we can get a `(Macro Syntax)` value out. The current code fails inside `expandEval` with `MacroEvaluationError`, whereas this branch fails here with `ValueNotMacro`. I think it should be `MacroEvaluationError`.
5. we do get a value out, but it's not a `(Macro Syntax)`, so `ValueNotMacro` is thrown. If this happens, that means there is a bug in Klister's type-checker, so `InternalError` or `error` would have been fine too.
6. `interpretMacroAction` encounters an error while executing the macro action. it fails somewhere inside `interpretMacroAction` via code which isn't relevant here.

As a conclusion, I think this should be `MacroEvaluationError`, not `ValueNotMacro`, but the existing code isn't very good at using the right error constructor, so this isn't a big deal.

Reply:
This is the same mistake as the other `erroneousValue` call site. Maybe `erroneousValue` shouldn't exist at all, since it seems so easy to misuse?